doc/src/build/dev_cheat_sheet.md

```diff
@@ -0,0 +1,41 @@
+# Sui Dev Cheat Sheet
+
+# Move
+
+### General
+
+- Read about [package upgrades](https://docs.sui.io/build/package-upgrades) and write upgrade-friendly code:
+    - Packages are immutable, so buggy package code can be called forever. Add protections at the object level instead.
+    - If you upgrade a package `P` to `P'`, other packages and clients that depend on `P` will continue using `P`, not auto-update to `P'`. Both dependent packages and client code must be explicitly updated to point at `P'`.
+    - `public` functions cannot be deleted or changed, but `public(friend)` functions can. Use `public(friend)` or private visibility liberally unless you are exposing library functions that will live forever.
+    - It is not possible to delete`struct` types, add new fields (though you can add [dynamic fields](https://docs.sui.io/devnet/build/programming-with-objects/ch5-dynamic-fields)), or add new [abilities](https://move-language.github.io/move/abilities.html) via an upgrade. Introduce new types liberally—they will live forever!
+- Use `vector`-backed collections (`vector`, `VecSet`, `VecMap`, `PriorityQueue`) with a **known** maximum size of ≤= 1000 items.
+    - Use dynamic field-backed collections (`Table`, `Bag`, `ObjectBag`, `ObjectTable`, `LinkedTable`) for larger collections, collections that contain an unbounded number of items, or any collection that allows third-party addition.
+    - Sui Move objects have a maximum size of 250KB—any attempt to create a larger object will lead to an aborted transaction. Ensure that your objects do not have an ever-growing `vector`-backed collection.
+    -
+- If your function `f` needs a payment in (e.g.) SUI from the caller, use `fun f(payment: Coin<SUI>)` not `fun f(payment: &mut Coin<SUI>, amount: u64)`. This is safer for callers—they know exactly how much they are paying, and do not need to trust `f` to extract the right amount.
+- Don’t micro-optimize gas usage. Sui computation costs are rounded up to the closest *[bucket](https://docs.sui.io/devnet/learn/tokenomics/gas-in-sui#gas-units)*, so only very drastic changes will make a difference. In particular, if your transaction is already in the lowest cost bucket, it can’t get any cheaper.
+- Follow the [Move coding conventions](https://move-language.github.io/move/coding-conventions.html) for consistent style.
+
+### Composability
+
+- Use the [`display`](https://docs.sui.io/build/sui-object-display) standard to customize how your objects show up in wallets, apps, and explorers
+- Avoid “self-transfers”—whenever possible, instead of writing `transfer::transfer(obj, tx_context::sender(ctx))`, return `obj` from the current function. This allows a caller or [programmable transaction block](https://docs.sui.io/build/prog-trans-ts-sdk) to use `obj`.
```

Comment:
```suggestion
- Avoid “self-transfers”. Whenever possible, instead of writing `transfer::transfer(obj, tx_context::sender(ctx))`, return `obj` from the current function. This allows a caller or [programmable transaction block](https://docs.sui.io/build/prog-trans-ts-sdk) to use `obj`.
```